### PR TITLE
Implement jprod and jtprod - creating the matrix

### PR DESCRIPTION
**Pull request by abelsiqueira**
Regarding JuliaOptimizers/NLPModels.jl#5

### CODE REVIEW

File: src/ampl_model.jl [hunk 1]
```diff
@@ -320,6 +321,44 @@ function jac(nlp :: AmplModel, x :: Array{Float64,1})
   return sparse(rows, cols, vals, nlp.meta.ncon, nlp.meta.nvar)
 end
 
+"""
+Evaluate the Jacobian-vector product at `x`.
+Warning: Currently building the Jacobian for this.
+"""
+function jprod(nlp :: AmplModel, x :: Array{Float64,1}, v :: Array{Float64,1})
+  return jac(nlp, x) * v
```
Comment (dpo) [inline src/ampl_model.jl]: I suggest `jprod` creates a vector and calls `jprod!` so the work is concentrated in one place.

File: src/ampl_model.jl [hunk 2]
```diff
@@ -320,6 +321,44 @@ function jac(nlp :: AmplModel, x :: Array{Float64,1})
...
+end
+
+"""
+Evaluate the transposed-Jacobian-vector product at `x` in place.
+Warning: Currently building the Jacobian for this.
+"""
+function jtprod!(nlp :: AmplModel,
+                x :: Array{Float64,1},
+                v :: Array{Float64,1},
+                Jtv :: Array{Float64,1})
```
Comment (dpo) [inline src/ampl_model.jl]: Indentation is a bit off here.

**Comment by abelsiqueira** [comment]
Updated

**Comment by dpo** [comment]
Thank you!

**Pull request merged**